### PR TITLE
New @ckeditor/ckeditor5-editor-classic v27

### PR DESCRIPTION
**Pull request by fedemp**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/ckeditor__ckeditor5-core/src/editor/editorconfig.d.ts [hunk 1]
```diff
@@ -88,14 +89,16 @@ export interface EditorConfig {
     mention?: MentionConfig;
     pagination?: PaginationConfig;
     placeholder?: string;
-    plugins?: Array<string | typeof Plugin>;
+    plugins?: Array<string | (new (editor: Editor) => Plugin)>;
```
Comment (peterblazejewicz) [inline types/ckeditor__ckeditor5-core/src/editor/editorconfig.d.ts:92]: 2C: this one should be probably exported as reusable type, describing a plugin class

**Review by peterblazejewicz — CHANGES_REQUESTED** [review]
Please review import of `EditorConfig`

File: types/ckeditor__ckeditor5-core/src/editor/editor.d.ts [hunk 2]
```diff
@@ -25,7 +25,7 @@ export default abstract class Editor implements Emitter, Observable {
     readonly state: "initializing" | "ready" | "destroyed";
 
     static builtinPlugins: Array<typeof Plugin>;
-    static defaultConfig: Record<string, unknown>;
+    static defaultConfig?: EditorConfig;
```
[the remark this answers is not in the document]
Comment (fedemp) [reply types/ckeditor__ckeditor5-core/src/editor/editor.d.ts:28]: Good catch.

File: types/ckeditor__ckeditor5-core/src/editor/editorconfig.d.ts (hunk 1 above)
Comment (fedemp) [reply types/ckeditor__ckeditor5-core/src/editor/editorconfig.d.ts]: I'll use `PluginInterface` that is exported from `core/plugin`.

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Pull request merged**